translate/set.go

```diff
@@ -105,6 +105,19 @@ func (ts TranslationSet) AddFromCommonSource(common path.ContextPath, toPrefix p
 	ts.AddTranslation(common, toPrefix)
 }
 
+// AddFromCommonObject adds translations for all of the paths in to from a matching path in from.
+// This is useful if both from and to have the same type (and does not make sense otherwise).
+func (ts TranslationSet) AddFromCommonObject(fromPrefix path.ContextPath, toPrefix path.ContextPath, to interface{}) {
+	vTo := reflect.ValueOf(to)
+	vPathsTo := prefixPaths(getAllPaths(vTo, ts.ToTag, true), toPrefix.Path...)
+	vPathsFrom := prefixPaths(getAllPaths(vTo, ts.FromTag, true), fromPrefix.Path...)
```

Comment:
This is probably okay as-is, but there's an unnecessary implicit assumption that `getAllPaths()` returns paths in a consistent order.  Possible cleanup: call `getAllPaths` once, drop the `prefixPaths` calls, and then loop over paths:

```go
ts.AddTranslation(prefixPath(path, fromPrefix.Path...), prefixPath(path, toPrefix.Path...))
```

Reply:
Followup in #357.